code/modules/power/lighting.dm

```diff
@@ -614,12 +616,25 @@
 	var/rigged = 0		// true if rigged to explode
 	var/brightness_range = 2 //how much light it gives off
 	var/brightness_power = 1
-	var/brightness_color = null
+	var/brightness_color
+	var/time_to_paint = 30  //amount of time it takes to paint a light
 
 /obj/item/light/ComponentInitialize()
 	. = ..()
 	AddComponent(/datum/component/caltrop, force)
 
+/obj/item/light/attackby(obj/item/O, mob/M)
+	..()
+	//color the light with a spraycan
+	if(istype(O, /obj/item/toy/crayon/spraycan) && do_after(M, time_to_paint, target = src))
+		var/obj/item/toy/crayon/spraycan/S = O
+		brightness_color =  S.colour
+		color = S.colour
+		visible_message("<span class='notice'>[M] paints the [src]!</span>")
```

Comment:
`the [src]` will turn out as `the the src` probably. You can leave the `the` out of it I think

Reply:
This is also missing a self message. Can be intended though